core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -83,7 +85,7 @@ class ReplicaFetcherThread(name: String,
 
   // process fetched data
   def processPartitionData(topicPartition: TopicPartition, fetchOffset: Long, partitionData: PartitionData) {
-    val replica = replicaMgr.getReplica(topicPartition).get
+    val replica = replicaMgr.getReplicaOrException(topicPartition)
```

Comment:
The test added in this patch will fail if we change this line to `val replica = replicaMgr.getReplica(topicPartition).get`